ui/packages/consul-ui/app/styles/base/color/README.mdx

```diff
@@ -0,0 +1,81 @@
+# Colors
+
+All colors for the immediate future should _mostly_ use the following `tone`
+CSS properties, unless:
+
+1. We get around to eventually adding semantic colors names (or adding something else) in which case these docs should be updated.
+2. You are absolutely sure the color you want should never be changed by a theme. These should be very very few and far between so even if we get that wrong it should be easy to update.
+
+`tones` should always using the following form:
+
+```css
+property: rgb(var(--tone-gray-500));
+```
+
+In other words always use `rgb` this gives us the flexibility to easily use thee variables with alpha if we ever need to.
+
+```css
+property: rgb(var(--tone-gray-500) / 50);
+```
+
+Currently, alphas are mainly used for shadows. Please consider whether you should be using an alpha before using an alpha'ed color.
```

Comment:
Good callout! Should we explicitly mention this is because it's important to avoid the use of alpha to simulate a tint or shade (especially because the effects can be unpredictable against backgrounds other than white)?

Reply:
```suggestion
Currently, alphas are mainly used for shadows. Please consider whether you should be using an alpha before using an alpha'ed color, you probably shouldn't be using an alpha, please use our collection of `--tones` instead for tints/shades of colors. If you do use one - you probably shouldn't :) , also expect it to be picked up in code review, so consider highlighting it in a self comment as you PR.
```

---

I'd actually like to get a linter involved a lot more on these things. We should be able to lint for alpha's and use ignores for the "once in a blue moon" occasions. With that and code review we have a nice balance of flexibility on the very rare occasions we need it and making it hard to do the wrong thing.

Reply:
> I'd actually like to get a linter involved a lot more on these things.

💯 for tooling on stuff like this.

Regarding the suggested change, if we are going to make adjustments to this text I'd like to focus on the "why" somebody should avoid using alphas here rather than repeatedly saying "you shouldn't do this" with increasing emphasis.

If it were me, I'd probably go with something like:

"Currently, alphas are mainly used for shadows. Please avoid the use of alphas for adjusting hues as they can produce unpredictable results. We should be utilizing our pre-defined color palette or adding new colors to it with proper RGB values. If you do need to add a color with alpha, please include a code comment or a self-comment explaining the use case."

Reply:
```suggestion
Currently, alphas are mainly used for shadows. Please avoid the use of alphas for adjusting hues as they can produce unpredictable results. We should be utilizing our pre-defined color palette or adding new colors to it with proper RGB values. If you do need to add a color with alpha, please include a code comment or a self-comment explaining the use case.
```